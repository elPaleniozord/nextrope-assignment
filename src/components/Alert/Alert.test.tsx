import React from 'react';
import { render } from '@testing-library/react';
import AlertPopup from './Alert';
import { MemoryRouter } from 'react-router';

test('should render alert success variant', () => {
  const { getByText, container } = render(
    <MemoryRouter>
      <AlertPopup status='success' />
    </MemoryRouter>
  );

  expect(getByText(/POWRÃ“T DO SKLEPU/)).toBeInTheDocument()
  
  expect(container).toMatchSnapshot()
});

test('should render alert warning variant', () => {
  const { getByText, container } = render(
    <AlertPopup status='Error message' />
  );

  expect(getByText(/ZAMKNIJ/)).toBeInTheDocument()
  
  expect(container).toMatchSnapshot()
});

test('should prevent render if status is undefined', () => {
  const { container}  = render(
    <AlertPopup status={undefined} />
  );
  
  expect(container.childElementCount).toBe(0)
});