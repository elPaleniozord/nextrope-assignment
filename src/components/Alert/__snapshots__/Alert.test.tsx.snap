// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render alert success variant 1`] = `
<div>
  <div
    class="fade d-flex flex-column align-items-center alert alert-success show"
    role="alert"
  >
    <div
      class="alert-heading h4"
    >
      Zamówienie Przyjęte
    </div>
    <p>
      Twoje zamówienie zostało przyjęte do realizacji.
    </p>
    <a
      class="nav-link"
      href="/"
    >
      <button
        class="btn btn-primary"
        type="button"
      >
        POWRÓT DO SKLEPU
      </button>
    </a>
  </div>
</div>
`;

exports[`should render alert warning variant 1`] = `
<div>
  <div
    class="fade d-flex flex-column align-items-center alert alert-danger show"
    role="alert"
  >
    <div
      class="alert-heading h4"
    >
      Błąd zamówienia
    </div>
    <p>
      Error message
    </p>
    <button
      class="btn btn-primary"
      type="button"
    >
      ZAMKNIJ
    </button>
  </div>
</div>
`;
